model/rockon.go

```diff
@@ -57,6 +57,7 @@ type Container struct {
 	Image        string                    `json:"image"`                   // docker image. eg: linuxserver/plex
 	Tag          string                    `json:"tag,omitempty"`           // tag of the docker image, if any. latest is used by default.
 	LaunchOrder  UintValue                 `json:"launch_order"`            // typically 1 or above. If there are multiple containers and they must be started in order, specify here.
+	uid          string                    `json:"uid,omitempty"`           // docker level UID:GUID or just UID (or -1 to invoke specific Rockstor logic), under which the container would run
```

Comment:
```suggestion
	Uid          *int32                    `json:"uid,omitempty"`           // invoke docker --user UID, or if -1 then first share owner.
```

Otherwise we get "Struct field 'uid' has 'json' tag but is not exported" The Capitalisation is important, and we have the json: tagging to indicate the actual field name. The large int type is required to include -1 and up to 65535 (UID max). And we must use a pointer to gain omitempty compatibility: Otherwise a value of zero would result in the entry being removed (some sense here as we default to 0, but less explicit).

Note also, from freshly created: https://github.com/rockstor/rockstor-core/issues/3029 that we reference here only a single integer, pertaining to the user only. A `gid` element additioin to our model, and it's implementation is sketched out in the last link.